pkg/controller/job/job_controller.go

```diff
@@ -792,12 +792,7 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (rErr error) {
 	var manageJobErr error
 	var finishedCondition *batch.JobCondition
 
-	jobHasNewFailure := failed > job.Status.Failed
-	// new failures happen when status does not reflect the failures and active
-	// is different than parallelism, otherwise the previous controller loop
-	// failed updating status so even if we pick up failure it is not a new one
-	exceedsBackoffLimit := jobHasNewFailure && (active != *job.Spec.Parallelism) &&
```

Comment:
I'm wondering why this was checked only when active != parallelism in the first place. Also, are there scenarios where the change manifests e2e? I suspend this change may conflict on cherry-picks (but not sure), so maybe can be done in a separate PR.

Reply:
This is necessary because a previous sync might have already set failed=backoffLimit.

Reply:
I honestly don't know why this check was so complex. We don't really need to know if the failure is new. All that matters is the backoffLimit

Reply:
sgtm